jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsEditor.kt

```diff
@@ -21,9 +21,12 @@ import javax.swing.JComponent
 
 class AwsConnectionSettingsEditor<T : AwsConnectionsRunConfigurationBase<*>>(
     project: Project,
-    private val settingsChangedListener: (AwsRegion?, String?) -> Unit = { _, _ -> }
+    settingsChangedListener: (AwsRegion?, String?) -> Unit = { _, _ -> }
 ) : SettingsEditor<T>() {
-    private val awsConnectionSelector = AwsConnectionSettingsSelector(project, settingsChangedListener)
+    private val awsConnectionSelector = AwsConnectionSettingsSelector(project) {
+        // TODO: Undo this unwrapping
```

Comment:
in part 3?

Reply:
Prob is own PR, it touches the run configs